defold-rive/pluginsrc/plugin.cpp

```diff
@@ -734,15 +741,40 @@ static void RiveEventCallback_Plugin(dmRive::RiveEventsContext* ctx)
                 // ro.m_SetBlendFactors = 1;
 
                 const rive::PaintData draw_entry_paint = rive::getPaintData(ctx->m_Paint);
-                const float* color                     = &draw_entry_paint.m_Colors[0];
 
-                ro.AddConstant(UNIFORM_COLOR, dmVMath::Vector4(color[0], color[1], color[2], color[3]));
+                dmVMath::Vector4 properties((float)draw_entry_paint.m_FillType, (float)draw_entry_paint.m_StopCount, 0.0f, 0.0f);
+                dmVMath::Matrix4 local_matrix;
+                dmRive::Mat2DToMat4(ctx->m_Event.m_TransformLocal, local_matrix);
+
+                dmVMath::Vector4 colors[rive::PaintData::MAX_STOPS];
+                for (int i = 0; i < (int) draw_entry_paint.m_StopCount; ++i)
+                {
+                    colors[i] = dmVMath::Vector4(draw_entry_paint.m_Colors[i*4+0],
+                                                 draw_entry_paint.m_Colors[i*4+1],
+                                                 draw_entry_paint.m_Colors[i*4+2],
+                                                 draw_entry_paint.m_Colors[i*4+3]);
+                }
+
+                dmVMath::Vector4 stops[rive::PaintData::MAX_STOPS];
+                for (int i = 0; i < (int) draw_entry_paint.m_StopCount; ++i)
+                {
+                    stops[i][0] = draw_entry_paint.m_Stops[i];
+                }
+
+                dmVMath::Vector4 gradient_limits(draw_entry_paint.m_GradientLimits[0], draw_entry_paint.m_GradientLimits[1], draw_entry_paint.m_GradientLimits[2], draw_entry_paint.m_GradientLimits[3]);
+
+                ro.AddConstant(UNIFORM_COLORS, colors, draw_entry_paint.m_StopCount);
+                ro.AddConstant(UNIFORM_TRANSFORM_LOCAL, (dmVMath::Vector4*) &local_matrix, 4);
+                ro.AddConstant(UNIFORM_GRADIENT_LIMITS, &gradient_limits, 1);
+                ro.AddConstant(UNIFORM_PROPERTIES, &properties, 1);
+                ro.AddConstant(UNIFORM_STOPS, stops, draw_entry_paint.m_StopCount);
```

Comment:
It's important to keep this implementation in sync with the runtime in comp_rive.cpp!